Code/GraphMol/SynthonSpaceSearch/SearchResults.h

```diff
@@ -11,19 +11,25 @@
 #ifndef RDKIT_SYNTHONSPACE_SEARCHRESULTS_H
 #define RDKIT_SYNTHONSPACE_SEARCHRESULTS_H
 
+#include <functional>
 #include <RDGeneral/export.h>
 #include <GraphMol/ROMol.h>
 
 namespace RDKit::SynthonSpaceSearch {
 
+// takes vector of search results; returns true if enough hits have been
+// returned, false if the search should continue.
+using SearchResultCallback =
+    std::function<bool(std::vector<std::unique_ptr<ROMol>> &)>;
```

Comment:
It seems like the callbacks shouldn't be modifying their arguments:
```suggestion
    std::function<bool(const std::vector<std::unique_ptr<ROMol>> &)>;
```
(this change requires other code changes)